### PR TITLE
Exorcises copy+paste from cult building code (minor cult building refactor / repathing)

### PR DESCRIPTION
**Pull request by MrMelbert**
## About The Pull Request

- Retyped the various cult buildings to be less vague / inconsistent. talisman -> altar, and tome -> archives.
- Removes a lot of copy+paste between the various subtypes of cult buildings. Now all cult buildings that produce items have a shared parent - `/cult/item_dispenser`. 
- Renamed a construct variable from `can_repair_constructs` to `can_repair`, to remove a hardcoded typecheck for artificers healing cult buildings.
- Library and Shuttle cult buildings used for decor are now their own subtypes, that don't drop runed metal on destruction. They do however still function for cultists, because why not
- An overall pass-over of the code to improve it to modern standards. Better var names, defines, etc

## Why It's Good For The Game

- More consistent behavior between cult buildings
- Cleaner, more modern code
- Allows for future cult prospects to be a lot less painless.

## Changelog

:cl: Melbert
refactor: Refactored cult buildings to exorcise some copy+paste. Some cosmetic cult buildings won't drop metal now (but still work for cultists!).
/:cl:

### CODE REVIEW

File: code/modules/antagonists/cult/cult_structure_archives.dm [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+			. += /obj/item/shuttle_curse
+		if(VEIL_WALKER)
+			. += /obj/item/cult_shift
+			. += /obj/item/flashlight/flare/culttorch
+
+/obj/structure/destructible/cult/item_dispenser/archives/succcess_message(mob/living/user, obj/item/spawned_item)
+	to_chat(user, span_cultitalic("You summon [spawned_item] from [src]!"))
+
+// Preset for the library that doesn't spawn runed metal on destruction.
+/obj/structure/destructible/cult/item_dispenser/archives/library
```
Comment (JohnFulpWillard) [inline code/modules/antagonists/cult/cult_structure_archives.dm:38]: why wouldnt the library one drop runed metal?

Comment (MrMelbert) [reply code/modules/antagonists/cult/cult_structure_archives.dm:38]: I changed the decor subtypes to not drop runed metal, as they were intended for decoration and not for people breaking them to fake cult

I'm not entirely sold on it, though

**Review by Maurukas — APPROVED** [review]

File: code/modules/antagonists/cult/cult_structure_altar.dm [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+		UNHOLY_WATER = image(icon = 'icons/obj/drinks.dmi', icon_state = "holyflask")
+		)
+	var/choice = show_radial_menu(user, src, items, custom_check = CALLBACK(src, .proc/check_menu, user), require_near = TRUE, tooltips = TRUE)
+	switch(choice)
+		if(ELDRITCH_WHETSTONE)
+			. += /obj/item/sharpener/cult
+		if(CONSTRUCT_SHELL)
+			. += /obj/structure/constructshell
+		if(UNHOLY_WATER)
+			. += /obj/item/reagent_containers/glass/beaker/unholywater
```
Comment (LemonInTheDark) [inline code/modules/antagonists/cult/cult_structure_altar.dm:29]: I feel like we could do this generically. List of define -> list(image, list(output) maybe?
Also you should make the radial options static

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]

File: code/modules/antagonists/cult/cult_structure_altar.dm (hunk 2 above)
Comment (MrMelbert) [reply code/modules/antagonists/cult/cult_structure_altar.dm]: for some reason i planned on making it static but never did
done now

**Pull request merged**